models/macros.yaml

```diff
@@ -10,7 +10,7 @@ macros:
       {% if warehouse.DatabaseType() == \"bigquery\" %}
         array_length( {{column_name}} )
       {% else %}
-        array_size( {{column_name}} )
+        array_size( parse_json({{column_name}}) )
```

Comment:
Won't it need parse_json in if-then as well as if-else?

Reply:
Now I remember, so the reason was that the macro array_size is not supposed to change the data type of inputs, if the data is already passed as an array then only it should work so caller should use parse_json and not the macro itself.

Reply:
But this macro  takes just the column name as input. Are you suggesting we need to have multiple macros one which takes the required data type while other that takes varchar?

Reply:
parse_json in if-then as well as if-else because this is not a safe method for Bigquery, it is used in else case because even if a column with array data type is passed then it won't harm but in BQ this will result in an error.